common/src/main/scala/org/apache/celeborn/common/metrics/source/ThreadPoolSource.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.metrics.source
+
+import java.util.concurrent.ThreadPoolExecutor
+
+import com.codahale.metrics.Gauge
+
+import org.apache.celeborn.common.CelebornConf
+
+class ThreadPoolSource(
+    threadPoolName: String,
+    threadPoolExecutor: ThreadPoolExecutor,
+    conf: CelebornConf,
+    role: String)
+  extends AbstractSource(conf, role) {
+  override val sourceName = s"THREAD_POOL_$threadPoolName"
+
+  addGauge(
+    s"${threadPoolName}_active_count",
```

Comment:
The source name has the thread pool name, therefore does this need to add the thread pool name in metric name?

Reply:
Move to label

Reply:
> The source name has the thread pool name, therefore does this need to add the thread pool name in metric name?

Changed